android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/NearbyBeaconsFragment.java

```diff
@@ -282,8 +282,12 @@ public void onListItemClick(ListView l, View v, int position, long id) {
     // Get the url for the given item
     PwPair item = mNearbyDeviceAdapter.getItem(position);
     if (!isFolderItem(item)) {
-      Intent intent = Utils.createNavigateToUrlIntent(item.getPwsResult());
-      startActivity(intent);
+      if (Utils.isWifiDirect(item.getUrlDevice())) {
+        new WifiDirectConnect(getActivity(), item.getUrlDevice());
```

Comment:
Add a comment.  I'm not sure why creating an object that immediately goes out of scope is helpful.